src/hooks/useActionCenterSubscriber/useSwapActionSubscriber.tsx

```diff
@@ -0,0 +1,412 @@
+import { getTxLink } from '@/lib/getTxLink'
+import { assertGetCosmosSdkChainAdapter } from '@/lib/utils/cosmosSdk'
+import { assertGetEvmChainAdapter } from '@/lib/utils/evm'
+import { assertGetSolanaChainAdapter } from '@/lib/utils/solana'
+import { assertGetUtxoChainAdapter } from '@/lib/utils/utxo'
+import { actionSlice } from '@/state/slices/actionSlice/actionSlice'
+import {
+  selectPendingSwapActions,
+  selectSwapActionBySwapId,
+} from '@/state/slices/actionSlice/selectors'
+import type { SwapAction } from '@/state/slices/actionSlice/types'
+import { ActionStatus, ActionType, isSwapAction } from '@/state/slices/actionSlice/types'
+import { selectFeeAssetByChainId } from '@/state/slices/selectors'
+import { swapSlice } from '@/state/slices/swapSlice/swapSlice'
+import { store, useAppDispatch, useAppSelector } from '@/state/store'
+
+type UseSwapActionSubscriberProps = {
+  onDrawerOpen: () => void
+}
+
+export const useSwapActionSubscriber = ({ onDrawerOpen }: UseSwapActionSubscriberProps) => {
+  const dispatch = useAppDispatch()
+  const translate = useTranslate()
+  const toastColor = useColorModeValue('white', 'gray.900')
+
+  const {
+    number: { toCrypto },
+  } = useLocaleFormatter()
+
+  const toast = useToast({
+    render: ({ title, status, description, onClose }) => {
+      const handleClick = () => {
+        onClose()
+        onDrawerOpen()
+      }
+
+      const toastSx = {
+        '&:hover': {
+          cursor: 'pointer',
+          background: status === 'success' ? 'green.300' : 'red.300',
+        },
+      }
+
+      return (
+        <Flex
+          // We can't memo this because onClose prop comes from the render props
+          // eslint-disable-next-line react-memo/require-usememo
+          onClick={handleClick}
+          background={status === 'success' ? 'green.500' : 'red.500'}
+          color='white'
+          px={4}
+          py={2}
+          borderRadius='md'
+          // eslint-disable-next-line react-memo/require-usememo
+          sx={toastSx}
+        >
+          <Box py={1} me={2}>
+            {status === 'success' ? (
+              <Icon color={toastColor} as={TbCircleCheckFilled} height='20px' width='20px' />
+            ) : (
+              <Icon color={toastColor} as={TbCircleXFilled} height='20px' width='20px' />
+            )}
+          </Box>
+          <Box>
+            <Text fontWeight='bold' color={toastColor}>
+              {title}
+            </Text>
+            <Text color={toastColor}>{description}</Text>
+          </Box>
+        </Flex>
+      )
+    },
+  })
+
+  const pendingSwapActions = useAppSelector(selectPendingSwapActions)
+  const swapsById = useAppSelector(swapSlice.selectors.selectSwapsById)
+  const {
+    state: { isConnected },
+  } = useWallet()
+  const activeSwapId = useAppSelector(swapSlice.selectors.selectActiveSwapId)
+  const previousSwapStatus = usePrevious(activeSwapId ? swapsById[activeSwapId]?.status : undefined)
```

Comment:
Not related to this PR but seeing we're still missing an idle state @NeOMakinG, do we want to bring this in in a follow-up?